staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -48,6 +48,11 @@ import (
 	utiltrace "k8s.io/utils/trace"
 )
 
+const (
+	// maxLimit is a maximum page limit increase used when fetching objects from etcd.
+	maxLimit = 10000
```

Comment:
what happens if an user request a limit greater than this? is that possible?

does this have user implications?

Reply:
Good point actually. The intent was not to touch any request with limit larger than maxLimit, but the implementation was different.

I changed the implementation to logic like:
* if limit < maxLimit => progressive resize 2x
* if limit >= maxLimit, keep the value, for backward compatibility

Re backward compatibility if limit >= maxLimit case: given experiments I made, I don't think performance of requests with limit >= 10k will differ from 10k, but I'm afraid there may be some tricky cases I'm not aware (like super tiny objects vs super large objects). To be on safe side, I would like to limit this change only to requests that are problematic (i.e. use small page size).

Reply:
I also added a comment.